src/vm/runners/cairo_runner.rs

```diff
@@ -237,6 +238,35 @@ impl CairoRunner {
         self.relocate_memory(&relocation_table);
         self.relocate_trace(&relocation_table);
     }
+
+    pub fn print_output(&self, stdout: &mut dyn io::Write) {
+        if let Some(builtin) = self.vm.builtin_runners.get("output") {
+            assert!(
```

Comment:
I have zero context but this looks suspicious: you are producing a runtime error if a method precondition is not met. Is there a way you can redesign the API such that it's impossible to print without computing effective sizes first?

Reply:
Yes, currently, compute_effective_sizes is called inside the relocate method, which is called before print_output, so the case handled by the assert would be unreachable, but it also makes sense to call compute_effective_sizes in print_output if it wasn't called before in order to unlink these two methods, as they perform independent tasks.